core-services/service-request/src/main/java/org/egov/servicerequest/validators/ServiceDefinitionRequestValidator.java

```diff
@@ -93,10 +87,49 @@ private void validateServiceDefinitionExistence(ServiceDefinition serviceDefinit
         }
     }
 
-    public void validateUpdateRequest(ServiceDefinitionRequest serviceDefinitionRequest) {
+    private List<ServiceDefinition> validateExistence(ServiceDefinition serviceDefinition) {
+        List<ServiceDefinition> serviceDefinitionList = serviceDefinitionRequestRepository.
+          getServiceDefinitions(ServiceDefinitionSearchRequest.builder().serviceDefinitionCriteria(ServiceDefinitionCriteria.builder().tenantId(serviceDefinition.getTenantId()).code(Collections.singletonList(serviceDefinition.getCode())).build()).build());
+        //check if valid service definition exists
+        if (CollectionUtils.isEmpty(serviceDefinitionList)) {
+            throw new CustomException(SERVICE_DEFINITION_NOT_EXIST_ERR_CODE, SERVICE_DEFINITION_NOT_EXIST_ERR_MSG);
+        }
+        //check if service definition is active or not
+        if (!serviceDefinitionList.get(0).getIsActive()) {
```

Comment:
Not required as search returns only active records. If this specific error must be thrown then add isActive false in the searchCriteria